### PR TITLE
Refactor network

### PR DESCRIPTION
**Pull request by IntelOrca**

### CODE REVIEW

**Comment by janisozaur** [comment]
```
/work/openrct2/src/interface/chat.c: In function ‘chat_draw’:
/work/openrct2/src/interface/chat.c:79:129: error: ‘NETWORK_AUTH_OK’ undeclared (first use in this function)
 status() != NETWORK_STATUS_CONNECTED || network_get_authstatus() != NETWORK_AUTH_OK) {
                                                                     ^~~~~~~~~~~~~~~
```

```
/home/travis/build/OpenRCT2/OpenRCT2/src/network/network.cpp:1433:9: error: 4 enumeration values not handled in switch: 'NETWORK_AUTH_NONE', 'NETWORK_AUTH_REQUESTED', 'NETWORK_AUTH_OK'... [-Werror,-Wswitch]
        switch(connection.AuthStatus) {
               ^
```

**Comment by IntelOrca** [comment]
welp

**Comment by IntelOrca** [comment]
Would someone be able to fix the xcode project?

File: src/network/network.cpp [hunk 1]
```diff
@@ -1988,79 +1483,79 @@ void Network::Server_Client_Joined(const char* name, const std::string &keyhash,
...
-			const std::string hash = connection.key.PublicKeyHash();
+		if (connection.AuthStatus == NETWORK_AUTH_VERIFIED) {
+			connection.AuthStatus = NETWORK_AUTH_OK;
+			const std::string hash = connection.Key.PublicKeyHash();
 			Server_Client_Joined(name, hash, connection);
 		} else
-		if (connection.authstatus != NETWORK_AUTH_REQUIREPASSWORD) {
-			log_error("Unknown failure (%d) while authenticating client", connection.authstatus);
+		if (connection.AuthStatus != NETWORK_AUTH_REQUIREPASSWORD) {
+			log_error("Unkown failure (%d) while authenticating client", connection.AuthStatus);
```
Comment (Nubbie) [inline src/network/network.cpp]: Spelling mistake: Unknown

**Comment by janisozaur** [comment]
OS X is failing *again* :P
```
❌  Undefined symbols for architecture i386

> Symbol: NetworkGroup::ToggleActionPermission(unsigned long)

> Referenced from: Network::LoadGroups() in network.o
```

**Comment by YJSoft** [comment]
0805f62ba6cc21ef84d4a7210f6691bed152265b is removed during rebase.

**Comment by janisozaur** [comment]
@YJSoft thanks for your prompt reply!

**Pull request merged**